### ISSUE TITLE
class balancing scheme

### ISSUE THREAD

**Issue by wzhouuu** [hellochick/ICNet-tensorflow#59]
Just wondering is there any class balancing schemes (e.g. in SegNet and ENet) used in this network to balance the small classes? I trained the model on cityscapes and then fine-tuned on my local data. But I found it has good results for large portion categories (sky, road, tree, etc.) but has bad results or no results for small portion categories (pedestrian, poles, etc.). Does anyone else have this problem or I might miss out something during training?

Also, another tensorflow question. Currently it saved models at every checkpoint. i would presume those saved models may not be the best models with lowest validation error. Has everyone organized to train the model towards a validation set but not at fixed checkpoint? 

Thank you for any help or hints!
